internal/advertisement-operator/config-watcher.go

```diff
@@ -4,15 +4,40 @@ import (
 	protocolv1 "github.com/liqoTech/liqo/api/advertisement-operator/v1"
 	policyv1 "github.com/liqoTech/liqo/api/cluster-config/v1"
 	"github.com/liqoTech/liqo/pkg/clusterConfig"
+	"github.com/liqoTech/liqo/pkg/crdClient"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog"
+	"time"
 )
 
-func (b *AdvertisementBroadcaster) WatchConfiguration(kubeconfigPath string) {
+func (b *AdvertisementBroadcaster) WatchConfiguration(kubeconfigPath string, client *crdClient.CRDClient) {
 	go clusterConfig.WatchConfiguration(func(configuration *policyv1.ClusterConfig) {
-		if configuration.Spec.AdvertisementConfig.ResourceSharingPercentage != b.ClusterConfig.ResourceSharingPercentage {
+		if !configuration.Spec.DiscoveryConfig.EnableAdvertisement {
 			klog.V(3).Info("ClusterConfig changed")
-			b.ClusterConfig.ResourceSharingPercentage = configuration.Spec.AdvertisementConfig.ResourceSharingPercentage
+			klog.Info("Stopping sharing resources with cluster " + b.ForeignClusterId)
+			err := b.NotifyAdvertisementDeletion()
```

Comment:
If advertisement does not exist in remote cluster this is a never ending process

If we don't find our Adv we can delete local PeeringRequest even if an error is returned